packages/terra-clinical-data-grid/src/DataGrid.jsx

```diff
@@ -932,16 +942,18 @@ class DataGrid extends React.Component {
         isSelectable={cell.isSelectable}
         isSelected={cell.isSelected}
         selectableRefCallback={(ref) => { this.cellRefs[cellKey] = ref; }}
+        isColumnHighlighted={isHighlighted}
+        isFirstRow={isFirstRow}
+        isLastRow={isLastRow}
       >
         {cell.component}
       </Cell>
     );
   }
 
-  renderRow(row, section, columns, width, isPinned, isStriped) {
+  renderRow(row, section, columns, width, isPinned, isStriped, isColumnHighlighted, isFirstRow, isLastRow) {
```

Comment:
Is the `isColumnHighlighted` argument here always going to be the same as `this.props.columnHighlightId` based on how it is passed down on line 1004? If so, can we just access it directly from `this.props.columnHighlightId` instead of passing it in as an argument?

Reply:
If the id is passed into the row, then the row has to determine if has a value or not inside the row component. It seems to make more sense to keep the evaluation outside of the row and in data-grid, so that all row needs to care about is a simple bool, similar to first/last row -- whatever the determining logic is to make it true/false (even though right now it is just checking if at least one column somewhere being highlighted). Keeps the row a little less knowledgable and less opinionated.  It's minor, but seems to be cleaner.

Reply:
Updated [here](https://github.com/cerner/terra-clinical/pull/796/commits/58bec54cb5437d28b3cc5f69cca242e6544f6f77#diff-412f754b40a1999c9f850f95d836fe5f21873db6a3feed84cbd813a65786e5ecR979)